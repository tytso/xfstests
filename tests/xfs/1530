#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-later
# Copyright (c) 2022 Oracle.  All Rights Reserved.
#
# FS QA Test No. 1530
#
# Populate a XFS filesystem and fuzz every refcountbt field.
# Try online repair and, if necessary, offline repair,
# to test the most likely usage pattern.

. ./common/preamble
_begin_fstest dangerous_fuzzers dangerous_bothrepair

_register_cleanup "_cleanup" BUS

# Import common functions.
. ./common/filter
. ./common/populate
. ./common/fuzzy
. ./common/reflink

# real QA test starts here
_supported_fs xfs
_require_scratch_reflink
_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

path="$(_scratch_xfs_find_agbtree_height 'refcnt' 2)" || \
	_fail "could not find two-level refcountbt"

echo "Fuzz refcountbt"
_scratch_xfs_fuzz_metadata '' 'both'  "$path" 'addr refcntroot' 'addr ptrs[1]' >> $seqres.full
echo "Done fuzzing refcountbt"

# success, all done
status=0
exit
