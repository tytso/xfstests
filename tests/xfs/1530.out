QA output created by 1530
Format and populate
Fuzz refcountbt
numrecs = lastbit: offline scrub didn't fail.
numrecs = lastbit: online scrub didn't fail.
leftsib = add: offline scrub didn't fail.
rightsib = ones: offline scrub didn't fail.
rightsib = middlebit: offline scrub didn't fail.
rightsib = lastbit: offline scrub didn't fail.
rightsib = add: offline scrub didn't fail.
Done fuzzing refcountbt
